server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -811,7 +812,7 @@ DataFileValue minorCompact(InMemoryMap memTable, TabletFile tmpDatafile, TabletF
         var storedFile = getDatafileManager().bringMinorCompactionOnline(tmpDatafile, newDatafile,
             new DataFileValue(stats.getFileSize(), stats.getEntriesWritten()), commitSession,
             flushId);
-        compactable.filesAdded(true, List.of(storedFile));
+        storedFile.ifPresent(stf -> compactable.filesAdded(true, List.of(stf)));
```

Comment:
Just to make sure I understand this original issue, was null being passed to List.of() what caused the problem?

Reply:
Yes. A NPE would be thrown. I don't know if it affected anything with compactions since it only happens when no file gets written.